### PR TITLE
[209] Add transition to suite selection for draws

### LINKED ISSUE #209: Admins can activate suite selection phase for draws
Once all groups have lottery numbers, admins can proceed to suite selection which will notify the first group(s) that they are eligible to select their suite.

### PR DESCRIPTION
**Pull request by orenyk**
Resolves #209
- Add `Draw#lottery_complete?`
- Fix bug with `Draw#ungrouped_students?` (it now only looks at undeclared and on_campus students
- Add start_selection route and `DrawSelectionStarter` service object

### CODE REVIEW

File: app/models/draw.rb [hunk 1]
```diff
@@ -69,7 +69,8 @@ def groups?
   #
   # @return [Boolean] whether or not there are any ungrouped students
   def ungrouped_students?
-    students.includes(:group).select { |s| s.group.nil? }.count.positive?
+    students.where(intent: %w(on_campus undeclared)).includes(:group)
+            .select { |s| s.group.nil? }.count.positive?
```
Comment (esoterik) [inline app/models/draw.rb]: should this still use the query object?

Comment (esoterik) [inline app/models/draw.rb]: also, undeclared students should be handled separately

Comment (orenyk) [inline app/models/draw.rb]: We're going to move this to #224, merging!

**Comment by orenyk** [comment]
Rebased, merging after green build (ignoring CodeClimate).

**Pull request merged**